ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+OpenSearch.swift

```diff
@@ -178,19 +178,20 @@ extension BrowserViewController {
     NetworkManager().downloadResource(with: url) { result in
       switch result {
       case .success(let response):
-        guard
-          let openSearchEngine = OpenSearchParser(pluginMode: true).parse(
-            response.data,
-            referenceURL: reference,
-            image: icon,
-            isCustomEngine: true
-          )
-        else {
-          return
+        Task { @MainActor in
+          guard
+            let openSearchEngine = await OpenSearchParser(pluginMode: true).parse(
```

Comment:
nit `if-let` instead of guard since there's only one statement after the guard block.

Reply:
This is prior code simply wrapped in a Task